docker-wsl-proxy.go

```diff
@@ -204,3 +212,112 @@ func handleContainerCreate(req *http.Request) error {
 
 	return nil
 }
+
+var containerInspectRegex = regexp.MustCompile("/containers/.*/json")
+
+func proxyModifyResponse(response *http.Response) (err error) {
+	if containerInspectRegex.MatchString(response.Request.RequestURI) && response.StatusCode == http.StatusOK {
+		err = handleContainerInspect(response)
+	}
+
+	return err
+}
+
+func handleContainerInspect(response *http.Response) error {
+	buf := make([]byte, response.ContentLength)
+	_, err := io.ReadFull(response.Body, buf)
+	if err != nil {
+		return err
+	}
+
+	containerJSON := types.ContainerJSON{}
+	err = json.Unmarshal(buf, &containerJSON)
+	if err != nil {
+		return err
+	}
+
+	for index, bind := range containerJSON.HostConfig.Binds {
+		containerJSON.HostConfig.Binds[index] = rewriteBindToWindows(bind)
+	}
+
+	for _, m := range containerJSON.Mounts {
+		if m.Type == mount.TypeBind {
+			m.Source = rewriteBindSourceToWindows(m.Source)
+		}
+	}
+
+	buf, err = json.Marshal(containerJSON)
+	if err != nil {
+		return err
+	}
+
+	response.Body = io.NopCloser(bytes.NewReader(buf))
+	response.ContentLength = int64(len(buf))
+	return err
+}
+
+func rewriteBindToWSL(bind string) string {
+	bind = path.Clean(bind)
+
+	var unixPrefixes = [...]string{"/host_mnt/", "/mnt/"}
+	for _, prefix := range unixPrefixes {
+		if strings.HasPrefix(bind, prefix) {
+			return "/mnt/" + bind[len(prefix):]
+		}
+	}
+
+	if len(bind) > 1 && bind[0] == '/' {
+		if bind[1] == ':' {
+			return bind
+		} else if len(bind) > 2 && bind[2] == ':' {
+			return "/mnt/" + strings.ToLower(bind[1:])
+		} else if bind[2] == '/' {
+			return "/mnt/" + strings.ToLower(bind[1:2]) + bind[2:]
+		} else {
+			return bind
+		}
+	}
+
+	// TODO: use NewLCOWParser from https://github.com/moby/moby/commit/28b0f47599e9ff32d2abebb54ee4b2a60977f722
+	var windowsBindParser = mounts.NewParser(mounts.OSLinux)
+
+	m, err := windowsBindParser.ParseMountRaw(bind, "")
+	if err != nil {
+		// Just hope that user is smarter than us
+		return bind
+	}
+	m.Source = strings.ReplaceAll(m.Source, `\`, `/`)
+
+	var driveSeps = [...]string{":/", "/"}
+	for _, driveSep := range driveSeps {
+		driveSepIdx := strings.Index(m.Source, driveSep)
+		if driveSepIdx >= 0 {
+			return "/mnt/" + strings.ToLower(m.Source[:driveSepIdx]) + "/" + m.Source[driveSepIdx+len(driveSep):] + bind[len(m.Source):]
```

Comment:
Пути более удобно цеплять через: `path.Join()`, например:
```suggestion
			return path.Join( "/mnt", strings.ToLower(m.Source[:driveSepIdx]), m.Source[driveSepIdx+len(driveSep):] + bind[len(m.Source):])
```